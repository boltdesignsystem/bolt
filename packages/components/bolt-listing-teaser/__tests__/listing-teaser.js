import { render, stopServer } from '../../../testing/testing-helpers';
const timeout = 90000;

describe('listing-teaser', () => {
  let page;

  afterAll(async () => {
    await stopServer();
    await page.close();
  });

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  test('basic usage', async () => {
    const results = await render(
      '@bolt-components-listing-teaser/listing-teaser.twig',
      {
        headline: {
          text: 'This is a listing-teaser',
          tag: 'h3',
          size: 'xlarge',
          link_attributes: {
            href: 'https://pega.com',
          },
        },
        meta_items: [
          'listing-teasered 8 hours 15 minutes ago',
          'Last activity: 2 minutes ago',
        ],
        status: {
          solved: true,
          solved_label: 'custom solved label',
          locked: true,
          locked_label: 'custom locked label',
          replies: '33',
          views: '123',
        },
        description: 'This is the description',
        warning: 'This is a warning message',
        attributes: {
          'data-foo': 'bar',
        },
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
