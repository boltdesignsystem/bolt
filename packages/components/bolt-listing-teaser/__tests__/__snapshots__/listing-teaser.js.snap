// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listing-teaser basic usage 1`] = `
<article data-foo="bar"
         class="c-bolt-listing-teaser js-test-target"
>
  <div class="c-bolt-listing-teaser__flag">
    <div class="c-bolt-listing-teaser__flag-heading">
      <h3 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
        <a href="https://pega.com"
           class="e-bolt-text-link e-bolt-text-link--reversed-underline e-bolt-text-link--expand-click-target"
        >
          This is a listing-teaser
        </a>
      </h3>
    </div>
    <div class="c-bolt-listing-teaser__flag-content">
      <ul class="c-bolt-listing-teaser__meta">
        <li class="c-bolt-listing-teaser__meta__item">
          listing-teasered 8 hours 15 minutes ago
        </li>
        <li class="c-bolt-listing-teaser__meta__item">
          Last activity: 2 minutes ago
        </li>
      </ul>
      <div class="c-bolt-listing-teaser__flag-content__item">
        This is the description
      </div>
      <div class="c-bolt-listing-teaser__flag-content__item c-bolt-listing-teaser__flag-content__item--warning">
        <bolt-icon name="warning"
                   color="yellow"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--warning  c-bolt-icon--yellow">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <g fill="none"
                 fill-rule="evenodd"
              >
                <path>
                </path>
                <g fill="var(--bolt-theme-icon, currentColor)">
                  <path d="M20.9 20H3a1 1 0 01-.9-1.6l9-14.8a1 1 0 011.7 0l8.9 14.8a1 1 0 01-1 1.6zm2.7-2.7l-9-14.8a3.1 3.1 0 00-5.3 0L.4 17.3a3.1 3.1 0 002.7 4.8H21a3.2 3.2 0 002.7-4.8zM12 7.4a1 1 0 00-1 1v4.2c0 .6.4 1 1 1s1-.4 1-1V8.4c0-.5-.4-1-1-1M12 15.8a1 1 0 100 2 1 1 0 000-2">
                  </path>
                </g>
              </g>
            </svg>
          </span>
        </bolt-icon>
        <div>
          This is a warning message
        </div>
      </div>
    </div>
    <ul class="c-bolt-listing-teaser__flag-status">
      <li class="c-bolt-listing-teaser__flag-status__item">
        <span class="c-bolt-listing-teaser__flag-status__icon c-bolt-listing-teaser__flag-status__icon--solved"
              aria-hidden="true"
        >
          <bolt-icon name="check-solid"
                     aria-hidden="true"
          >
            <span class="c-bolt-icon c-bolt-icon--check-solid">
              <svg class="c-bolt-icon__icon"
                   xmlns="http://www.w3.org/2000/svg"
                   data-name="Layer 1"
                   viewbox="0 0 22 22"
              >
                <path fill="var(--bolt-theme-icon, currentColor)"
                      d="M11 0a11 11 0 1011 11A11 11 0 0011 0zm5.2 8.3L10 14.4a1 1 0 01-1.5 0l-2.8-2.8a1 1 0 010-1.5 1 1 0 011.5 0l2 2 5.4-5.3a1 1 0 01.7-.3 1 1 0 01.8.3 1 1 0 010 1.5z"
                >
                </path>
              </svg>
            </span>
          </bolt-icon>
        </span>
        custom solved label
      </li>
      <li class="c-bolt-listing-teaser__flag-status__item">
        <span class="c-bolt-listing-teaser__flag-status__icon c-bolt-listing-teaser__flag-status__icon--locked"
              aria-hidden="true"
        >
          <bolt-icon name="lock"
                     aria-hidden="true"
          >
            <span class="c-bolt-icon c-bolt-icon--lock">
              <svg class="c-bolt-icon__icon"
                   xmlns="http://www.w3.org/2000/svg"
                   viewbox="0 0 24 24"
              >
                <g fill="none"
                   fill-rule="evenodd"
                >
                  <path>
                  </path>
                  <path fill="var(--bolt-theme-icon, currentColor)"
                        d="M20 20c0 .6-.4 1-1 1H5a1 1 0 01-1-1v-7c0-.6.5-1 1-1h14c.6 0 1 .4 1 1v7zM8 7a4 4 0 018 0v3H8V7zm11 3h-1V7A6 6 0 006 7v3H5a3 3 0 00-3 3v7a3 3 0 003 3h14a3 3 0 003-3v-7a3 3 0 00-3-3z"
                  >
                  </path>
                </g>
              </svg>
            </span>
          </bolt-icon>
        </span>
        custom locked label
      </li>
      <li class="c-bolt-listing-teaser__flag-status__item">
        <span class="c-bolt-listing-teaser__flag-status__number">
          33
        </span>
        Replies
      </li>
      <li class="c-bolt-listing-teaser__flag-status__item">
        <span class="c-bolt-listing-teaser__flag-status__number">
          123
        </span>
        Views
      </li>
    </ul>
  </div>
</article>
`;
